godot-core/src/engine/script_instance.rs

```diff
@@ -219,10 +148,11 @@ type ScriptInstanceInfo = sys::GDExtensionScriptInstanceInfo;
 type ScriptInstanceInfo = sys::GDExtensionScriptInstanceInfo2;
 
 struct ScriptInstanceData<T: ScriptInstance> {
-    inner: RefCell<T>,
+    inner: Pin<Box<GdCell<T>>>,
```

Comment:
Is this triple nesting truly necessary?

Reply:
that is the type `GdCell::new(...)` returns. [`InstanceStorage`](https://github.com/godot-rust/gdext/blob/7482941eed1637329641bd466e64fc0bacf3f8fc/godot-core/src/storage/single_threaded.rs#L16-L23) uses the same type.

Reply:
I see, let's leave it for now. I really don't like the ergonomics around `Pin` but I guess that ship in Rust has sailed...

Reply:
It hasn't quite sailed, there are proposals that could improve/replace it eventually. But yeah it is needed since we do need *some* way to pin the `GdCell` after construction and the easiest is to pin a boxed value.